include/cppkafka/buffer.h

```diff
@@ -92,6 +92,17 @@ class CPPKAFKA_API Buffer {
         static_assert(sizeof(T) == sizeof(DataType), "sizeof(T) != sizeof(DataType)");
     }
 
+    /**
+     * Constructs a buffer from an array
+     *
+     * \param data The the array to be used as input
+     */
+    template <typename T, size_t N>
+    Buffer(const std::array<T, N>& data)
+    : data_(reinterpret_cast<const DataType*>(&data[0])), size_(N * sizeof(T)) {
```

Comment:
Minor: can you use `data.data()` and `data.size()` rather than `&data[0]` and `N * sizeof(T)` here?

Reply:
will do! first time working with std::arrays

Reply:
Perfect, that works! One more thing: can you add a deleted constructor taking a `std::array<T, N>&&` just like there is for `vector` and `string`? e.g. [this one](https://github.com/mfontanini/cppkafka/blob/master/include/cppkafka/buffer.h#L97). This will make sure people can't misuse `Buffer` and construct it from a temporary `array`, which will end up making the buffer contain garbage once it goes out of scope.

Reply:
should be fixed